cmake/SetupCodeMetrics.cmake

```diff
@@ -49,7 +49,28 @@
 # Note: Only supported for gnu or clang.
 ########################################
 if (ENABLE_COVERAGE)
-    include(blt/cmake/CodeCoverage.cmake)
+    ##########################################################################
+    # Setup coverage compiler flags 
+    ##########################################################################
+    # Set the actual flags for coverage in the COVERAGE_FLAGS variable 
+    # Note: '--coverage' is equivalent to '-fprofile-arcs -ftest-coverage' for compiling and '-lgcov' for linking
+    # Additional flags that might be useful: 
+    #       " -fno-inline -fno-inline-small-functions -fno-default-inline"
+    blt_append_custom_compiler_flag(FLAGS_VAR   COVERAGE_FLAGS 
+                                    DEFAULT     "--coverage")
```

Comment:
The old logic simply error when it wasn't gcc and used the `DEFAULT` path.

I can make that change.

It looks like we use DEFAULT, for gcc?

What else should I add?  I see cases like this for XL, are there existing examples for `INTEL`